File: rocket_app/app/models/alert.rb
# == Schema Information
#
# Table name: alerts
#
#  id            :bigint           not null, primary key
#  errorId       :string           not null
#  errorSeverity :string           not null
#  errorCategory :string           not null
#  errorMessage  :string           not null
#  longMessage   :string           not null
#  errorTime     :bigint           not null
#  selected      :boolean          not null
#  new           :boolean          not null
#  expanded      :boolean          not null
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#
class Alert < ApplicationRecord
  validates :errorId, :errorSeverity, :errorCategory, :errorMessage, :longMessage, :errorTime, presence: true
end

File: backend/model/alert.js
const mongoose = require('mongoose');

const alertSchema = new mongoose.Schema({
  errorId: { type: String },
  errorSeverity: { type: String },
  errorCategory: { type: String },
  errorMessage: { type: String },
  longMessage: { type: String },
  errorTime: { type: Number },
  selected: { type: Boolean },
  new: { type: Boolean },
  expanded: { type: Boolean }
});

module.exports = mongoose.model('alert', alertSchema);
